lib/gooddata/lcm/actions/synchronize_users.rb

```diff
@@ -226,74 +230,71 @@ def call(params)
-                        c = domain.clients(client_id)
-                        if params.segments_filter && !segments_filter.include?(c.segment_uri)
-                          puts "Client #{client_id} is outside segments_filter #{params.segments_filter}"
+                        segment_client = domain_clients.find { |domain_client| domain_client.client_id.to_s == client_id.to_s }
+                        if params.segments_filter && !segments_filter.include?(segment_client.segment_uri)
+                          semaphore.synchronize { params.gdc_logger.warn("Client #{client_id} is outside segments_filter #{params.segments_filter}") }
                           next
                         end
-                        project = c.project
-                        fail "Client #{client_id} does not have project." unless project
+                        client_project = segment_client.project
+                        fail "Client #{client_id} does not have project." unless client_project
                         working_client_ids << client_id.to_s
-                        puts "Project #{project.pid} of client #{client_id} will receive #{users.count} users"
-                        project.import_users(users,
-                                             domain: domain,
-                                             whitelists: whitelists,
-                                             ignore_failures: ignore_failures,
-                                             remove_users_from_project: remove_users_from_project,
-                                             do_not_touch_users_that_are_not_mentioned: do_not_touch_users_that_are_not_mentioned,
-                                             create_non_existing_user_groups: create_non_existing_user_groups)
+                        semaphore.synchronize { params.gdc_logger.info("Project #{client_project.pid} of client #{client_id} will receive #{users.count} users") }
+                        client_project.import_users(users,
+                                                    domain: domain,
+                                                    whitelists: whitelists,
+                                                    ignore_failures: ignore_failures,
+                                                    remove_users_from_project: remove_users_from_project,
+                                                    do_not_touch_users_that_are_not_mentioned: do_not_touch_users_that_are_not_mentioned)
                       end
 
                       params.gdc_logger.debug("Working client ids are: #{working_client_ids.join(', ')}")
 
                       unless do_not_touch_users_that_are_not_mentioned
-                        domain_clients.each do |c|
-                          next if working_client_ids.include?(c.client_id.to_s)
+                        domain_clients.peach do |domain_client|
+                          next if working_client_ids.include?(domain_client.client_id.to_s)
```

Comment:
Identical code found in synchronize_user_filters.rb:190, synchronize_users.rb:276 (mass = 88)